superset/sql_parse.py

```diff
@@ -1622,7 +1622,7 @@ def extract_tables_from_jinja_sql(sql: str, database: Database) -> set[Table]:
     return (
         tables
         | ParsedQuery(
-            sql_statement=processor.process_template(template),
+            sql_statement=sql,
```

Comment:
@leoguo1024 we purposefully don't wont to render the Jinja template here as it might reference the `[hive|presto|trino].latest_partition` macro (or similar), i.e., the failing test a good check to verify the existing logic.